### PR TITLE
fix(stage, deps): use de-angularized provider service, remove babel

### PR DESCRIPTION
**Pull request by erikmunson**
The result of an exciting roller coaster of a journey...

So, turns out the `kayentaCanary` stage config UI has been broken when running against any `@spinnaker/core` since (I think?) `0.0.360`. https://github.com/spinnaker/deck/pull/6953 converted `ProviderSelectionService` to react, and Kayenta's usage didn't update so we've been blowing up.

Separately, we're behind on `core` by like 30 versions. Upgrading to the necessary version managed to also take in changes to some components that we use in Kayenta's unit tests which got changed to use the new React context API. Apparently the version of `enzyme` that we're on doesn't support that, so I had to bump that and related deps to match. _But then_ they didn't work because that introduced a change into how yarn picked the version of `semver` to use, so I had to pin that to one that all the libs were happy with.

And finally, something really incredibly wonky happened and the babel-transformed output of async/await calls stopped working (stopped loading in the `regenerator` shims). I assumed it was something from #429, but I reversed the dep updates I made and same result. Perhaps it also snuck in via the `core` bump. We're likely about to move to a simpler build process and babel doesn't have a great reason to be in picture these days, so I opted to just take it out of the build entirely rather than try to keep limping along the complex web of build dependencies.

And now we're here! Hopefully this breaks fewer things than it fixes 😆

### CODE REVIEW

File: build_scripts/webpack.build.js [hunk 1]
```diff
@@ -75,25 +73,19 @@ module.exports = {
   },
   plugins: [
     new ForkTsCheckerWebpackPlugin({ checkSyntacticErrors: true }),
-    new webpack.optimize.UglifyJsPlugin({
```
Comment (erikmunson) [inline build_scripts/webpack.build.js]: this plugin is chronically unhappy with the output of `ts-loader`. I imagine it's also not entirely necessary given that folks can/will be minifying their app builds.

**Comment by maggieneterval** [comment]
That roller coaster of a journey sounds...nauseating. Much appreciated, Erik! :pray:

**Comment by erikmunson** [comment]
@maggieneterval Just imagine how much fun you'd miss out on if I stopped constantly banging my head against figurative walls every time I try to touch something

**Comment by christopherthielen** [comment]
wowza

**Pull request merged**